core/scripts/cre/environment/environment/billing.go

```diff
@@ -0,0 +1,241 @@
+}
+
+func removeBillingStateFiles(relativePathToRepoRoot string) error {
+	path := filepath.Join(relativePathToRepoRoot, envconfig.StateDirname, envconfig.BillingStateFilename)
+	absPath, absErr := filepath.Abs(path)
+	if absErr != nil {
+		return errors.Wrap(absErr, "error getting absolute path for billing platform service state file")
+	}
+
+	return os.Remove(absPath)
+}
+
+func startBilling(_ context.Context, cleanupWait time.Duration, setupOutput *environment.SetupOutput) (startupErr error) {
+	// just in case, remove the stack if it exists
+	_ = framework.RemoveTestStack(billingplatformservice.DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME)
+
+	defer func() {
+		p := recover()
+
+		if p != nil {
+			fmt.Println("Panicked when starting Billing Platform Service")
+
+			if err, ok := p.(error); ok {
+				fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+				fmt.Fprintf(os.Stderr, "Stack trace: %s\n", string(debug.Stack()))
+
+				startupErr = err
+			} else {
+				fmt.Fprintf(os.Stderr, "panic: %v\n", p)
+				fmt.Fprintf(os.Stderr, "Stack trace: %s\n", string(debug.Stack()))
+
+				startupErr = fmt.Errorf("panic: %v", p)
+			}
+
+			time.Sleep(cleanupWait)
+
+			billingRemoveErr := framework.RemoveTestStack(billingplatformservice.DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME)
+			if billingRemoveErr != nil {
+				fmt.Fprint(os.Stderr, errors.Wrap(billingRemoveErr, manualBillingCleanupMsg).Error())
+			}
+
+			os.Exit(1)
+		}
+	}()
+
+	stageGen := stagegen.NewStageGen(1, "STAGE")
+	fmt.Print(libformat.PurpleText("%s", stageGen.Wrap("Starting Billing Platform Service")))
+
+	previousCTFConfig := os.Getenv("CTF_CONFIGS")
+	defer func() {
+		setErr := os.Setenv("CTF_CONFIGS", previousCTFConfig)
+		if setErr != nil {
+			framework.L.Warn().Msgf("failed to restore previous CTF_CONFIGS environment variable: %s", setErr)
+		}
+	}()
+
+	setErr := os.Setenv("CTF_CONFIGS", DefaultBillingConfigFile)
+	if setErr != nil {
+		return fmt.Errorf("failed to set CTF_CONFIGS environment variable: %w", setErr)
+	}
+
+	// Load and validate test configuration
+	in, err := framework.Load[envconfig.BillingConfig](nil)
+	if err != nil {
+		return errors.Wrap(err, "failed to load test configuration")
+	}
+
+	if setupOutput != nil {
+		in.BillingService.ChainSelector = setupOutput.WorkflowRegistryConfigurationOutput.ChainSelector
+		addressRefs, err := setupOutput.CldEnvironment.DataStore.Addresses().Fetch()
+		if err != nil {
+			return errors.Wrap(err, "failed to fetch address references")
+		}
+
+		for _, ref := range addressRefs {
+			switch ref.Type {
+			case "WorkflowRegistry":
+				if in.BillingService.ChainSelector == ref.ChainSelector {
+					in.BillingService.WorkflowRegistryAddress = ref.Address
+				}
+			case "CapabilitiesRegistry":
+				if in.BillingService.ChainSelector == ref.ChainSelector {
+					in.BillingService.CapabilitiesRegistryAddress = ref.Address
+				}
+			default:
+				continue
+			}
```

Comment:
how will you handle multiple versions of this appearing?

Reply:
Ufda, to start, the data available to different components in the local cre vs the test environments are very strange and not clear to work with. I would love to clean this up, but I'm still not certain of the expected data pattern or how we're supposed to access certain values. :(